extension/src/pipeline/quickPick.ts

```diff
@@ -160,28 +172,29 @@ const pickFields = async (
     return
   }
 
-  const yValues = (await quickPickUserOrderedValues(
-    items.filter(
-      item => item.value === undefined || !xValues?.includes(item.value)
-    ),
-    {
-      title: Title.SELECT_PLOT_Y_METRIC
-    }
-  )) as { file: string; field: string }[] | undefined
+  const xValuesLength = xValues?.length || 0
 
-  const y = verifyYFields(yValues, xValues?.length)
+  const yItems = items.filter(
+    item => item.value === undefined || !xValues?.includes(item.value)
+  )
+  const y =
+    xValuesLength > 1
+      ? await pickYFieldsWithMultiXFields(yItems, xValuesLength)
+      : await pickYFields(yItems)
```

Comment:
With this change, y fields are picked and validated differently depending on if there are multiple x values.